backend/src/impl/default_controllers_impl.py

```diff
@@ -220,7 +237,27 @@ def benchmark_get_by_id(benchmark_id: str, by_creator: bool) -> Benchmark:
 
 
 def systems_get_by_id(system_id: str) -> System:
-    return SystemDBUtils.find_system_by_id(system_id)
+    system = SystemDBUtils.find_system_by_id(system_id)
+    user = get_user()
+    if system.is_private:
+        if user.is_authenticated:
+            if _is_creator(system, user):
+                return system
+            elif _is_shared_user(system, user):
+                # must be creator to see shared_users
+                system.shared_users = None
+                return system
+            else:
+                abort_with_error_message(403, "system access denied", 40302)
+        else:
+            abort_with_error_message(403, "system access denied", 40302)
```

Comment:
I think this can be either but maybe a 401 makes more sense here. What do you think?

Reply:
Ah right, unloggedin users should receive 401 instead of 403. 
But now that both the loggedin check and permission check are included in _has_read_access(system), I'm not sure what's the best status code... perhaps 403?

Reply:
Yeah, 403 sounds good!